rosapi/src/rosapi/proxy.py

```diff
@@ -218,20 +218,17 @@ def get_subscribers(topic, topics_glob):
         return []
 
 
-def get_service_providers(servicetype, services_glob):
+def get_service_providers(queried_type, services_glob):
     """ Returns a list of node names that are advertising a service with the specified type """
-    try:
-        if any_match(str(topic), services_glob):
-            publishers, subscribers, services = Master('/rosbridge').getSystemState()
-            servdict = dict(services)
-            if servicetype in servdict:
-                return servdict[servicetype]
-            else:
-                return []
-        else:
-            return []
-    except socket.error:
-        return []
+    _, _, services = Master('/rosbridge').getSystemState()
```

Comment:
Hi Jihoon,

No, I had actually missed that. But I noticed something else while looking at it now. I realized that the `getSystemState()`'s documentation says:
```
Raises:
    rosgraph.masterapi.Error - if Master returns ERROR.
    rosgraph.masterapi.Failure - if Master returns FAILURE.
```
While we're excpecting a `socket.error` the except blocks of all similar calls. Am I missing something? because neither of them is a subclass of `socket.error`.